code/game/objects/items/stacks/rods.dm

```diff
@@ -73,3 +73,12 @@ var/global/list/datum/stack_recipe/rod_recipes = list ( \
 
 /obj/item/stack/rods/cyborg/update_icon()
 	return
+
+/obj/item/stack/rods/ten
+	amount = 10
+
+/obj/item/stack/rods/twentythree

```

Comment:
it does exactly what you've done here, it's a compile time creation of a new type.
that's it, think of them as anonymous types/classes, they're defined and used once (unless stored in a var)